kythe/java/com/google/devtools/kythe/platform/java/helpers/BlockAnonymousSignatureGenerator.java

```diff
@@ -179,7 +179,13 @@ public Void visitClass(ClassTree classTree, BlockAnonymousData blockData) {
     JCClassDecl classDecl = (JCClassDecl) classTree;
     if (classTree.getSimpleName().contentEquals("")) {
       StringBuilder anonymousClassSignature = new StringBuilder();
-      anonymousClassSignature.append(getBlockSignature(classDecl.sym));
+      if (classDecl.sym != null) {
+        anonymousClassSignature.append(getBlockSignature(classDecl.sym));
+      } else {
+        String name = classDecl == null ? "" : classDecl.name.toString();
```

Comment:
You check that classDecl is null here, but not above on line 182.

Reply:
should have been checking for name